From patchwork Tue Dec  4 22:39:33 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: "tip-bot for Paul E. McKenney" <tipbot@zytor.com>
X-Patchwork-Id: 10712645
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 8BB9B13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Dec 2018 22:39:43 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 72D532BF8E
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Dec 2018 22:39:43 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 65E6B2BF9F; Tue,  4 Dec 2018 22:39:43 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 1403D2BF8E
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Dec 2018 22:39:43 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726358AbeLDWjm (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 4 Dec 2018 17:39:42 -0500
Received: from terminus.zytor.com ([198.137.202.136]:36003 "EHLO
        terminus.zytor.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725875AbeLDWjm (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 4 Dec 2018 17:39:42 -0500
Received: from terminus.zytor.com (localhost [127.0.0.1])
        by terminus.zytor.com (8.15.2/8.15.2) with ESMTPS id wB4MdXf21037295
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=NO);
        Tue, 4 Dec 2018 14:39:33 -0800
Received: (from tipbot@localhost)
        by terminus.zytor.com (8.15.2/8.15.2/Submit) id wB4MdX4p1037292;
        Tue, 4 Dec 2018 14:39:33 -0800
Date: Tue, 4 Dec 2018 14:39:33 -0800
X-Authentication-Warning: terminus.zytor.com: tipbot set sender to
 tipbot@zytor.com using -f
From: "tip-bot for Paul E. McKenney" <tipbot@zytor.com>
Message-ID: <tip-d05faa5f1ac50beef77b4ceba0e8e157d41146e2@git.kernel.org>
Cc: virtualization@lists.linux-foundation.org, mst@redhat.com,
        tglx@linutronix.de, kvm@vger.kernel.org, paulmck@linux.ibm.com,
        hpa@zytor.com, mingo@kernel.org, netdev@vger.kernel.org,
        jasowang@redhat.com
Reply-To: tglx@linutronix.de, linux-kernel@vger.kernel.org, mst@redhat.com,
          virtualization@lists.linux-foundation.org, jasowang@redhat.com,
          netdev@vger.kernel.org, mingo@kernel.org, paulmck@linux.ibm.com,
          hpa@zytor.com, kvm@vger.kernel.org
To: linux-tip-commits@vger.kernel.org
Subject: [tip:core/rcu] drivers/vhost: Replace synchronize_rcu_bh() with
 synchronize_rcu()
Git-Commit-ID: d05faa5f1ac50beef77b4ceba0e8e157d41146e2
X-Mailer: tip-git-log-daemon
Robot-ID: <tip-bot.git.kernel.org>
Robot-Unsubscribe: Contact <mailto:hpa@kernel.org> to get blacklisted from
 these emails
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Disposition: inline
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Commit-ID:  d05faa5f1ac50beef77b4ceba0e8e157d41146e2
Gitweb:     https://git.kernel.org/tip/d05faa5f1ac50beef77b4ceba0e8e157d41146e2
Author:     Paul E. McKenney <paulmck@linux.ibm.com>
AuthorDate: Mon, 5 Nov 2018 17:14:53 -0800
Committer:  Paul E. McKenney <paulmck@linux.ibm.com>
CommitDate: Tue, 27 Nov 2018 09:21:37 -0800

drivers/vhost: Replace synchronize_rcu_bh() with synchronize_rcu()

Now that synchronize_rcu() waits for bh-disable regions of code as well
as RCU read-side critical sections, synchronize_rcu_bh() can be replaced
by synchronize_rcu().  This commit therefore makes this change.

Signed-off-by: Paul E. McKenney <paulmck@linux.ibm.com>
Cc: "Michael S. Tsirkin" <mst@redhat.com>
Cc: Jason Wang <jasowang@redhat.com>
Cc: <kvm@vger.kernel.org>
Cc: <virtualization@lists.linux-foundation.org>
Cc: <netdev@vger.kernel.org>
---
 drivers/vhost/net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/vhost/net.c b/drivers/vhost/net.c
index ab11b2bee273..564ead864028 100644
--- a/drivers/vhost/net.c
+++ b/drivers/vhost/net.c
@@ -1359,7 +1359,7 @@ static int vhost_net_release(struct inode *inode, struct file *f)
 	if (rx_sock)
 		sockfd_put(rx_sock);
 	/* Make sure no callbacks are outstanding */
-	synchronize_rcu_bh();
+	synchronize_rcu();
 	/* We do an extra flush before freeing memory,
 	 * since jobs can re-queue themselves. */
 	vhost_net_flush(n);
